low-level/sensors/impl/centos_7/service_monitor.py

```diff
@@ -0,0 +1,490 @@
+            # Unit is not found for the service
+            services_to_monitor_copy = self.services_to_monitor.copy()
+            for service in services_to_monitor_copy:
+                err = self.connect_to_prop_changed_signal(service)
+                if err:
+                    self.raise_alert(service, "N/A", "N/A", "N/A", "N/A",
+                        "N/A", "N/A", 0)
+                    logger.error(f"{service} is not active initially. \n Error {err}")
+                else:
+                    self.services_to_monitor.remove(service)
+
+            logger.debug(f"failed_services : {self.failed_services}")
+            logger.debug(f"services_to_monitor : {self.services_to_monitor}")
+
+            # Retrieve the main loop which will be called in the run method
+            self._loop = GLib.MainLoop()
+
+            # Initialize the gobject threads and get its context
+            GLib.threads_init()
+            context = self._loop.get_context()
+
+            time_to_check_lists = current_time() + self.polling_frequency
+
+            while self.is_running():
+                # At interval of 'thread_sleep' check for events occured for
+                # registered services and process them(call on_pro_changed())
+                context.iteration(False)
+                time.sleep(self.thread_sleep)
+
+                # At interval of 'polling_freqency' process unregistered
+                # services and services with not-active state.
+                if time_to_check_lists <= current_time():
+                    time_to_check_lists = current_time() + \
+                                            self.polling_frequency
+
+                    # Try to bind the enabled services on the node to the
+                    # signal whose Unit was earlier not found. On successfully
+                    # registering for service state change signal, remove from
+                    # local list as monitoring enabled through SystemD
+                    # and to avoid re-registration.
+                    services_to_monitor_copy = self.services_to_monitor.copy()
+                    for service in services_to_monitor_copy:
+                        if not self.connect_to_prop_changed_signal(service):
+                            self.services_to_monitor.remove(service)
+
+                    # Check for services in intermidiate state(not active)
+                    self.check_notactive_services()
+
+
+            logger.info("ServiceMonitor gracefully breaking out " +\
+                                "of dbus Loop, not restarting.")
+        except GLib.Error as err:
+            raise ThreadException(self.SENSOR_NAME,
+                "Ungrecefully breaking out of GLib.MainLoop() with error: %s"
+                %err)
+        except DBusException as err:
+            raise ThreadException(self.SENSOR_NAME,
+                "Ungracefully breaking out of dbus loop with error: %s"% err)
+        except Exception as err:
```

Comment:
Unknown exception would be caught by sspl-ll-d generic exception handling, however that would halt the service itself. Instead by raising ThreadException specifically, only this thread would halted, so main & other sensor threads could continue. 
With thsi understanding, i feel this is fine.